crates/fluvio-spu/src/services/public/mod.rs

```diff
@@ -1,30 +1,143 @@
 mod api_versions;
-mod service_impl;
 mod produce_handler;
 mod fetch_handler;
 mod offset_request;
+mod offset_update;
 mod stream_fetch;
 
-use tracing::info;
+use std::sync::Arc;
+use async_trait::async_trait;
+use tracing::{info, debug, trace, instrument};
+use futures_util::StreamExt;
 
-use fluvio_service::FluvioApiServer;
-use service_impl::PublicService;
+use fluvio_socket::{FluvioSocket, SocketError};
+use fluvio_service::{FluvioApiServer, FluvioService, ConnectInfo, call_service};
 use fluvio_spu_schema::server::SpuServerRequest;
 use fluvio_spu_schema::server::SpuServerApiKey;
+use fluvio_types::event::StickyEvent;
 
 use crate::core::DefaultSharedGlobalContext;
+use self::api_versions::handle_api_version_request;
+use self::produce_handler::handle_produce_request;
+use self::fetch_handler::handle_fetch_request;
+use self::offset_request::handle_offset_request;
+use self::offset_update::handle_offset_update;
+use self::stream_fetch::StreamFetchHandler;
 pub use stream_fetch::publishers::StreamPublishers;
 
-pub(crate) type PublicApiServer =
-    FluvioApiServer<SpuServerRequest, SpuServerApiKey, DefaultSharedGlobalContext, PublicService>;
+pub(crate) type SpuPublicServer = FluvioApiServer<
```

Comment:
Adding SPU prefix  is redundant.  We are in the SPU so there is no confusion here.

Reply:
They are different because they are import from another modules.  If it is renamed, we have rename other variable as well.

Reply:
Do you mean if the `SpuPublicServer` type is renamed? I would say the same thing about the `create_public_service` function, I don't think this argument favors one over the other.

Reply:
Not seeing any increase in clarity or reduction of code with proposed renaming other than making everything verbose.  Just stick with changes that improve instrumentation or debugging.